src/runtime/wasm_math.ll

```diff
@@ -140,28 +140,15 @@ define weak_odr <8 x i16> @saturating_narrow_i32x8_to_u16x8(<8 x i32> %x) nounwi
 }
 
 ; Integer to double-precision floating point
+define weak_odr <4 x double> @i32_to_double_s(<4 x i32> %x) nounwind alwaysinline {
+  %1 = sitofp <4 x i32> %x to <4 x double>
```

Comment:
This is almost certainly what CodeGen_LLVM does. Are you sure we need these helpers?

Reply:
Without these helpers, everything gets scalarized.

Reply:
(But yeah, this is definitely suboptimal. I'll follow up with wasm LLVM folks to see if this can be improved.)

Reply:
Why does it scalarize? That doesn't make any sense.

Reply:
You are correct. It doesn't make any sense. I'm contacting the relevant wasm person right now.

Reply:
Looking more closely at the code, these cases actually *don't* get scalarized, but neither to they generate the opcodes we expect; instead, it generates a sequence of ops that loads two int32s, extends them to int64, then does bit-twiddling to convert them to IEEE754 doubles. Maybe there's a codegen path that predates these instructions being added to the wasmSimd128 spec that hasn't been updated? I'll circle around with the wasm-llvm folks to see what's going on. In the meantime, this is arguably an improvement on the status quo.

Reply:
I'm going to leave this open (not landed) pending the LLVM investigation.

Reply:
Did you also report this for float -> double, from #6132?

Reply:
I'm going to provide repro cases for that as well

Reply:
Opened https://bugs.llvm.org/show_bug.cgi?id=51098